src/protocols/secure_channel/tests/TestCheckinMsg.cpp

```diff
@@ -253,6 +268,110 @@ void TestCheckin_GenerateParse(nlTestSuite * inSuite, void * inContext)
     }
 }
 
+/**
+ * @brief Test verifies that the nonce generation is successful when using valid inputs
+ */
+void TestCheckInMsg::TestCheckInMessageNonceGeneration(nlTestSuite * inSuite, void * inContext)
+{
+    TestSessionKeystoreImpl keystore;
+
+    int numOfTestCases = ArraySize(checkIn_message_test_vectors);
+    for (int numOfTestsExecuted = 0; numOfTestsExecuted < numOfTestCases; numOfTestsExecuted++)
+    {
+        CheckIn_Message_test_vector vector = checkIn_message_test_vectors[numOfTestsExecuted];
+
+        uint8_t buffer[300] = { 0 };
+        Encoding::LittleEndian::BufferWriter writer(buffer, sizeof(buffer));
+
+        Symmetric128BitsKeyByteArray hmacKeyMaterial;
+        memcpy(hmacKeyMaterial, vector.key, vector.key_len);
+
+        Hmac128KeyHandle hmac128KeyHandle;
+        NL_TEST_ASSERT_SUCCESS(inSuite, keystore.CreateKey(hmacKeyMaterial, hmac128KeyHandle));
+
+        // Verify that the generation succeeded
+        NL_TEST_ASSERT_SUCCESS(inSuite, CheckinMessage::GenerateCheckInMessageNonce(hmac128KeyHandle, vector.counter, writer));
```

Comment:
Just make this method public. In fact, since it doesn't rely on any class state at all, couldn't it just be a function in CheckinMessage.h?

Reply:
Refactored tests to validate nonce generation with the public geneartion APIs